src/DCAttribute.cpp

```diff
@@ -20,77 +20,68 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-
-
-#include <sstream>
-
 #include "splash/core/DCAttribute.hpp"
+#include "splash/core/H5IdWrapper.hpp"
+#include "splash/DCAttributeInfo.hpp"
+
+#include <cassert>
 
 namespace splash
 {
-
-    DCAttribute::DCAttribute()
-    {
-    }
-
     std::string DCAttribute::getExceptionString(const char *name, std::string msg)
     {
         return (std::string("Exception for DCAttribute [") + std::string(name) +
                 std::string("] ") + msg);
     }
 
-    void DCAttribute::readAttribute(const char* name, hid_t parent, void* dst)
+    DCAttributeInfo* DCAttribute::readAttributeInfo(const char* name, hid_t parent)
     throw (DCException)
     {
         hid_t attr = H5Aopen(parent, name, H5P_DEFAULT);
-        if (attr < 0)
+        if (!attr)
```

Comment:
Jep, this is a bug. Formerly it was an H5AttributeId which has a `!operator` that does the checking